mypy/subtypes.py

```diff
@@ -463,8 +463,12 @@ def visit_instance(self, left: Instance) -> bool:
                         assert unpacked.type.fullname == "builtins.tuple"
                         if isinstance(get_proper_type(unpacked.args[0]), AnyType):
                             return not self.proper_subtype
-            # TODO: we need a special case similar to above to consider (something that maps to)
-            # tuple[Any, ...] a subtype of Tuple[<whatever>].
+                if (
+                    mapped.type.has_base("builtins.tuple")
```

Comment:
Why did you need the last commit? This condition is always true because of the surrounding `if` (each tuple fallback has `builtins.tuple` in MRO), and also now this check looks wrong because it may trigger in unrelated cases involving something like `class M(Generic[T], tuple[int, ...]): ...`.

What you had before was almost right, except you didn't need the check `len(mapped.args) == 1`, it should be always true if `mapped.type.fullname == "builtins.tuple"`.

Reply:
I misinterpreted your comment here https://github.com/python/mypy/pull/16108#discussion_r1325700221
Ugh, I tried to test the `class M(Generic[T], tuple[int, ...]): ...` case but I did it backwards. Will fix and add test cases!

Reply:
If you will not be able to come up with a test case, don't try to hard. It may be not easy to trigger such edge case (because of how user defined tuple types are represented). You already have enough test cases :-)

(I would still remove the unneeded len check)